packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -56,14 +95,31 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
             initialSummary,
             wipSummaryLogger,
         );
+
+        this.initialGCDetailsP = new LazyPromise(async () => {
+            return getInitialGCDetailsFn ? getInitialGCDetailsFn() : undefined;
+        });
     }
 
     public async summarize(fullTree: boolean, trackState: boolean = true): Promise<IContextSummarizeResult> {
+        // Update the reference used state from the initial GC details. This is used to find out if the used state has
+        // changed from the last state seen by the server. If so, we cannot reuse previous summary.
+        if (this.referenceUsedState === undefined) {
+            this.referenceUsedState = (await this.initialGCDetailsP)?.used;
+        }
+
         // If trackState is true, get summary from base summarizer node which tracks summary state.
         // If trackState is false, get summary from summarizeInternal.
         if (trackState) {
             const summarizeResult = await super.summarize(fullTree);
 
+            if (!this.used && this.referenceUsedState !== this.used) {
```

Comment:
Isn't it the same as 
 if (!this.used && this.referenceUsedState) {

Reply:
If this.referencedState is undefined, then also we would want to write the used state. Unless we are making the default value of initial used state to be true as discussed in one of the earlier comments.

Reply:
Updated this as per the changes where we we are storing used routes.